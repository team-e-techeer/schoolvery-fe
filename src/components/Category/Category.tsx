import colors from '@/constants/colors';
import { css } from '@emotion/react';
import { Link } from 'react-router-dom';

export interface ImgInfo {
  path: string;
  src: string;
  name: string;
}

interface Props {
  imageInfo: ImgInfo[];
}

const imageWrapper = css`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

export default function Category({ imageInfo }: Props) {
  return (
    <>
      <div
        css={css`
          display: flex;
          flex-direction: column;
          width: 90%;
          margin: 2rem auto;
          background-color: ${colors.grey100};
          border-radius: 8px;
          padding: 2rem;
          max-height: 30rem;
          box-shadow: 1px 1px 1px 1px ${colors.grey300};
        `}
      >
        <ul css={imageWrapper}>
          {imageInfo.map((image, idx) => (
            <Link
              css={css`
                display: flex;
                flex-direction: column;
                flex-basis: 20%;
                flex-grow: 0;
                flex-shrink: 1;
                align-items: center;
                align-content: center;
                max-height: 10rem;
              `}
              key={idx}
              to={image.path}
            >
              <img
                css={css`
                  max-width: 100%;
                  max-height: 7rem;
                `}
                src={image.src}
                alt={`category img for ${image.name}`}
              />
              <span>{image.name}</span>
            </Link>
          ))}
        </ul>
      </div>
    </>
  );
}
