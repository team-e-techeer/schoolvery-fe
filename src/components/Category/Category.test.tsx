import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter, Router } from 'react-router-dom';
import Category from './Category';
import type { ImgInfo } from './Category';
import Ch from '../../assets/img/ch.png';
import Chicken from '../../assets/img/Chicken.png';
import Jp from '../../assets/img/jp.png';
import Ko from '../../assets/img/ko.png';
import { createMemoryHistory, MemoryHistory } from 'history';
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('로그인 페이지 테스트', () => {
  it('렌더링 테스트', () => {
    const history: MemoryHistory = createMemoryHistory();
    const imageInfos: ImgInfo[] = [
      { path: '/category/korea', src: Ko, name: '한식' },
      { path: '/category/china', src: Ch, name: '중식' },
      { path: '/category/japan', src: Jp, name: '일식' },
      { path: '/category/chicken', src: Chicken, name: '치킨' },
    ];

    const { container } = render(
      <Router location={history.location} navigator={history}>
        <Category imageInfo={imageInfos} />
      </Router>
    );
    history.push('/');
    expect(screen.getByAltText('category img for 한식')).toBeInTheDocument();
    expect(screen.getByAltText('category img for 중식')).toBeInTheDocument();
    expect(screen.getByAltText('category img for 일식')).toBeInTheDocument();
    expect(screen.getByAltText('category img for 치킨')).toBeInTheDocument();
    expect(history.location.pathname).toBe('/');
    const toKorea = screen.getByText('한식');
    fireEvent.click(toKorea);
    expect(history.location.pathname).toBe('/category/korea');
    expect(container).toMatchSnapshot();
  });
});
