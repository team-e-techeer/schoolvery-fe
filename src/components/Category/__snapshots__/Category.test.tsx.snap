// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`로그인 페이지 테스트 렌더링 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 90%;
  margin: 2rem auto;
  background-color: #f2f4f6;
  border-radius: 8px;
  padding: 2rem;
  max-height: 30rem;
  box-shadow: 1px 1px 1px 1px #d1d6db;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-basis: 20%;
  -ms-flex-preferred-size: 20%;
  flex-basis: 20%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  max-height: 10rem;
}

.emotion-3 {
  max-width: 100%;
  max-height: 7rem;
}

<div>
  <div
    class="emotion-0"
  >
    <ul
      class="emotion-1"
    >
      <a
        class="emotion-2"
        href="/category/korea"
      >
        <img
          alt="category img for 한식"
          class="emotion-3"
          src="test-file-stub"
        />
        <span>
          한식
        </span>
      </a>
      <a
        class="emotion-2"
        href="/category/china"
      >
        <img
          alt="category img for 중식"
          class="emotion-3"
          src="test-file-stub"
        />
        <span>
          중식
        </span>
      </a>
      <a
        class="emotion-2"
        href="/category/japan"
      >
        <img
          alt="category img for 일식"
          class="emotion-3"
          src="test-file-stub"
        />
        <span>
          일식
        </span>
      </a>
      <a
        class="emotion-2"
        href="/category/chicken"
      >
        <img
          alt="category img for 치킨"
          class="emotion-3"
          src="test-file-stub"
        />
        <span>
          치킨
        </span>
      </a>
    </ul>
  </div>
</div>
`;
